spec/01-unit/03-conf_loader_spec.lua

```diff
@@ -670,6 +670,44 @@ describe("Configuration loader", function()
           assert.is_nil(err)
           assert.same(http_tls.modern_cipher_list, conf.ssl_ciphers)
         end)
+        it("defines ssl_protocols by default", function()
+          local conf, err = conf_loader(nil, {})
+          assert.is_nil(err)
+          -- looks kinda like a cipher suite
+          assert.equal("TLSv1.1 TLSv1.2 TLSv1.3", conf.ssl_protocols, nil, true)
+        end)
+        it("explicitly defines ssl_protocols", function()
+          local conf, err = conf_loader(nil, {
+            ssl_protocols = "tlsv1"
+          })
+          assert.is_nil(err)
+          -- looks kinda like a cipher suite
+          assert.equal("TLSv1", conf.ssl_protocols, nil, true)
+        end)
+        it("explicitly defines multiple ssl_protocols (comma separated)", function()
+          local conf, err = conf_loader(nil, {
+            ssl_protocols = "SSLV3, tlsv1"
+          })
+          assert.is_nil(err)
+          -- looks kinda like a cipher suite
+          assert.equal("SSLv3 TLSv1", conf.ssl_protocols, nil, true)
+        end)
+        it("explicitly defines multiple ssl_protocols (space separated)", function()
+          local conf, err = conf_loader(nil, {
+            ssl_protocols = "SSLV3  tlsv1"
+          })
+          assert.is_nil(err)
+          -- looks kinda like a cipher suite
+          assert.equal("SSLv3 TLSv1", conf.ssl_protocols, nil, true)
+        end)
+        it("doesn't define ssl_protocols when no protocols specified (uses nginx defaults)", function()
```

Comment:
In this test case it seems we fall back to the default value of http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_protocols, which defines TLS v1 (unlike our current posture where we set a minimum protocol version of 1.1). Do we want to stick with the existing default, rather than the nginx defaults (such a change would probably catch folks deploying 1.2 off guard if they regularly test their handshake capabilities)

Reply:
Existing default is still used as default. To fallback to Nginx default you need to actively ask for it.